app/routes.py

```diff
@@ -0,0 +1,345 @@
+        return jsonify(None), 400
+    if customer is None:
+        return jsonify(None), 404
+
+    video = Video.query.get(video_id)
+    if video_id is None:
+        return jsonify(None), 400
+    if video is None:
+        return jsonify(None), 404
+
+    rental = Rental.query.filter_by(customer_id=customer_id, video_id=video_id, checked_in_status=False).order_by(Rental.due_date.asc()).first()
+    if rental is None:
+        return jsonify({"message" : f"No outstanding rentals for customer {customer_id} and video {video_id}"
+        }), 400
+
+    rental.checked_in_status = True
+    video_checked_out_count = Rental.query.filter_by(video_id=video_id, checked_in_status=False).count()
+    available_inventory = video.total_inventory - video_checked_out_count
+
+    db.session.commit()
+
+    return {
+        "customer_id": rental.customer_id,
+        "video_id": rental.video_id,
+        "videos_checked_out_count": video_checked_out_count,
+        "available_inventory": available_inventory 
+        }, 200
+
+@customers_bp.route("/<customer_id>/rentals", methods=["GET"])
+def list_customer_rentals(customer_id):
+    customer = Customer.query.get(customer_id)
+    if customer is None:
+        return jsonify({
+            "message" : f"Customer {customer_id} was not found"
+        }), 404
+    customer_rentals = Rental.query.filter_by(customer_id=customer_id, checked_in_status=False).all()
+    customer_rentals_response = []
+    for customer_rental in customer_rentals:
+        video = Video.query.get(customer_rental.video_id)
+        customer_rentals_response.append({
+            "release_date" : video.release_date,
+            "title" : video.title,
+            "due_date" : customer_rental.due_date
+        })
+    return jsonify(customer_rentals_response)
+
+
+@videos_bp.route("/<video_id>/rentals", methods=["GET"])
+def list_video_rentals(video_id):
+    video = Video.query.get(video_id)
+    if video is None:
+        return jsonify({
+            "message" : f"Video {video_id} was not found"
+        }), 404
+    video_rentals = Rental.query.filter_by(video_id=video_id, checked_in_status=False).all()
+    video_rentals_response = []
+    for video_rental in video_rentals:
+        customer = Customer.query.get(video_rental.customer_id)
+        video_rentals_response.append({
+            "due_date" : video_rental.due_date,
+            "name" : customer.name,
+            "phone" : customer.phone,
+            "postal_code" : customer.postal_code
+        })
+    return jsonify(video_rentals_response)
+
+
+@videos_bp.route("", methods=["GET", "POST"])
+def handle_videos():
+    if request.method == "GET":
+        videos = Video.query.all()
+        videos_response = []
+        for video in videos:
+            videos_response.append({
+                "id": video.id,
+                "title": video.title,
+                "release_date": video.release_date,
+                "total_inventory": video.total_inventory
```

Comment:
There are many places in our routes where we need to build a dictionary like this (or very similar). Consider making a helper method as an instance method of the video class (e.g. `def to_dict(self)`).